### PR TITLE
Add NSPasteboard and associated APIs

### PR DESCRIPTION
**Pull request by burtonageo**
Adds:

- NSPasteboard
- NSPasteboardItem
- NSPasteboardReading
- NSPasteboardWriting
- NSPasteboardReadingOptions
- NSPasteboardWritingOptions

---
This change is [Reviewable](https://reviewable.io/reviews/servo/cocoa-rs/134)

### CODE REVIEW

File: src/appkit.rs [hunk 1]
```diff
@@ -381,6 +392,285 @@ impl NSRunningApplication for id {
...
+impl NSPasteboardReading for id {
+    unsafe fn initWithPasteboardPropertyList_ofType(self, propertyList: id, _type: id) -> id {
+        msg_send![self, initWithPasteboardPropertyList:propertyList ofType:_type]
+    }
+
+    unsafe fn readableTypesForPasteboard(self, pasteboard: id) -> id {
+        let class: id = msg_send![self, class];
+        msg_send![class, readableTypesForPasteboard:pasteboard]
+    }
+    unsafe fn readingOptionsForType(self, pasteboard: id)  {
```
Comment (jdm) [inline src/appkit.rs]: https://developer.apple.com/reference/appkit/nspasteboardreading/1533559-readingoptionsfortype?language=objc says this should return a value.

**Comment by jdm** [comment]
This generally looks good, but please squash the commits about typos and compile errors - there's no need for them to appear in the master commit history.

**Comment by burtonageo** [comment]
All done, thanks :-)

File: src/appkit.rs [hunk 2]
```diff
@@ -381,6 +392,285 @@ impl NSRunningApplication for id {
...
+impl NSPasteboardReading for id {
+    unsafe fn initWithPasteboardPropertyList_ofType(self, propertyList: id, _type: id) -> id {
+        msg_send![self, initWithPasteboardPropertyList:propertyList ofType:_type]
+    }
+
+    unsafe fn readableTypesForPasteboard(self, pasteboard: id) -> id {
+        let class: id = msg_send![self, class];
+        msg_send![class, readableTypesForPasteboard:pasteboard]
+    }
+    unsafe fn readingOptionsForType_pasteboard(self, _type: id, pasteboard: id) -> id {
```
Comment (burtonageo) [inline src/appkit.rs]: Oops, fixed now

**Comment by jdm** [comment]
@bors-servo: r+
Thanks!

**Pull request merged**